utils/forest_shim/src/sector.rs

```diff
@@ -0,0 +1,271 @@
+    ) -> Self {
+        SectorInfo(SectorInfoV3 {
+            proof,
+            sector_number,
+            sealed_cid,
+        })
+    }
+}
+
+impl Deref for SectorInfo {
+    type Target = SectorInfoV3;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<SectorInfo> for SectorInfoV2 {
+    fn from(value: SectorInfo) -> SectorInfoV2 {
+        SectorInfoV2 {
+            proof: RegisteredSealProof(value.0.proof).into(),
+            sealed_cid: value.sealed_cid,
+            sector_number: value.sector_number,
+        }
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq)]
+pub struct RegisteredPoStProof(RegisteredPoStProofV3);
+
+impl Deref for RegisteredPoStProof {
+    type Target = RegisteredPoStProofV3;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<RegisteredPoStProofV3> for RegisteredPoStProof {
+    fn from(value: RegisteredPoStProofV3) -> Self {
+        RegisteredPoStProof(value)
+    }
+}
+
+impl Inner for RegisteredPoStProof {
+    type FVM = RegisteredPoStProofV3;
+}
+
+impl From<RegisteredPoStProofV2> for RegisteredPoStProof {
+    fn from(value: RegisteredPoStProofV2) -> RegisteredPoStProof {
+        let proof = match value {
+            RegisteredPoStProofV2::StackedDRGWinning2KiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWinning2KiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWinning8MiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWinning8MiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWinning512MiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWinning512MiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWinning32GiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWinning32GiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWinning64GiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWinning64GiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWindow2KiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWindow2KiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWindow8MiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWindow8MiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWindow512MiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWindow512MiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWindow32GiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWindow32GiBV1
+            }
+            RegisteredPoStProofV2::StackedDRGWindow64GiBV1 => {
+                RegisteredPoStProofV3::StackedDRGWindow64GiBV1
+            }
+            RegisteredPoStProofV2::Invalid(i64) => RegisteredPoStProofV3::Invalid(i64),
+        };
+        RegisteredPoStProof(proof)
+    }
+}
+
+/// SectorSize indicates one of a set of possible sizes in the network.
+#[derive(Clone, Debug, PartialEq, Copy, serde::Serialize)]
+
```

Comment:
Do we need `#[serde(transparent)]` here?

Reply:
no, removed

Reply:
Why don't we need it?

Reply:
oh wait, i removed the one on `PoStProof`.  For sector size, I think we would need repr_transparent as the underying type has `repr(u64)` and `Se(de)rialize_repr` attributes. i'll add them just to be safe.

Reply:
The need for `#[serde(transparent)]` is completely separate from the representation, though. We should add tests to make sure we get the serialization right.

Reply:
I've added tests and it appears that #[serde(transparent)] is not needed.